src/views/login.tsx

```diff
@@ -1,35 +1,220 @@
 import { useLazyQuery } from "@apollo/client";
-import React from "react";
+import React, { useState } from "react";
 
 import queries from "../settings/graphql-queries";
 
+import styled from "styled-components";
+import {
+    OuterContainer,
+    OuterForm,
+    FormInner,
+    Header,
+    Body,
+    Button,
+    SubText,
+    InputEl,
+    ForgotPasswordLink,
+} from "../styles/CreateAccountElements";
+import { ErrorMessage } from "../styles/ErrorMessageElements";
+import Logo from "../assets/logo-coloured.png";
+import MailLock from "../assets/mail-lock.png";
+import PassLock from "../assets/lock.png";
+import { useNavigate } from "react-router-dom";
+
 const LoginView = () => {
+    let navigate = useNavigate();
     const [getSignIn, signInResult] = useLazyQuery(queries.signIn);
     const [isSignedIn, setIsSignedIn] = React.useState(false);
     React.useEffect(() => {
-        if(signInResult){
-            if(!signInResult.loading)
-            {
-                if(signInResult.error) console.error(signInResult.error)
+        if (signInResult) {
+            if (!signInResult.loading) {
+                if (signInResult.error) console.error(signInResult.error);
                 else if (signInResult.data) {
-                    console.log(signInResult.data.signIn)
-                    setIsSignedIn(signInResult.data.signIn)
+                    console.log(signInResult.data.signIn);
+                    setIsSignedIn(signInResult.data.signIn);
                 }
             }
             console.log(signInResult);
             // TODO:
             // SetContextAuthenticated(true);
         }
-    }, [signInResult])
-    
-    
+    }, [signInResult]);
+
+    const [formState, setFormState] = useState(false);
+    const [errors, setErrors] = useState<{
+        password?: string;
+        email?: string;
+    }>({});
+    const [inputs, setInputs] = useState<{
+        password?: string;
+        email?: string;
+    }>({});
+
+    const useForm = (initialValues: any) => {
+        const handleSubmit = (e: any) => {
+            if (e) {
+                e.preventDefault();
+            }
+            console.log(inputs);
+        };
+        const handleInputChange = (e: any) => {
+            e.persist();
+            setInputs((inputs) => ({
+                ...inputs,
+                [e.target.name]: e.target.value,
+            }));
+        };
+        return {
+            handleSubmit,
+            handleInputChange,
+            inputs,
+        };
+    };
+
+    const validate = (inputs: any) => {
+        const cond = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(
```

Comment:
This is kinda scarey but tbh I don't care anymore about email validation using regex

Reply:
I'll remove it :D I don't think it's neccessary anyway with this project.